src/module.c

```diff
@@ -505,8 +511,28 @@ static int internalAdd(RedisModuleCtx *ctx,
         return REDISMODULE_ERR;
     }
 
+    // Use module level configuration if key level configuration doesn't exists
+    DuplicatePolicy dp_policy;
+    if (dp_override != DP_NONE) {
+        dp_policy = dp_override;
+    } else if (series->duplicatePolicy != DP_NONE) {
+        dp_policy = series->duplicatePolicy;
+    } else {
+        dp_policy = TSGlobalConfig.duplicatePolicy;
+    }
+
+    // Insert filter for close samples. If configured, it's used to ignore last measurement if its
+    // value is negligible compared to the last sample.
+    if (dp_policy == DP_LAST && series->srcKey == NULL && series->totalSamples != 0 &&
+        timestamp >= series->lastTimestamp &&
+        timestamp - series->lastTimestamp <= series->ignoreMaxTimeDiff &&
+        fabs(value - series->lastValue) <= series->ignoreMaxValDiff) {
```

Comment:
Maybe it would be faster to avoid fabs and use
((value - series->lastValue) <= series->ignoreMaxValDiff) && 
 (value - series->lastValue) >= -series->ignoreMaxValDiff)       )